spec/Section 5 -- Validation.md

```diff
@@ -1517,6 +1528,125 @@ query ($foo: Boolean = true, $bar: Boolean = false) {
 }
 ```
 
+### Defer And Stream Directives Are Used On Valid Root Field
+
+** Formal Specification **
+
+- For every {directive} in a document.
+- Let {directiveName} be the name of {directive}.
+- Let {mutationType} be the root Mutation type in {schema}.
+- Let {subscriptionType} be the root Subscription type in {schema}.
+- If {directiveName} is "defer" or "stream":
+  - The parent type of {directive} must not be {mutationType} or
+    {subscriptionType}.
+
+**Explanatory Text**
+
+The defer and stream directives are not allowed to be used on root fields of the
+mutation or subscription type.
+
+For example, the following document will not pass validation because `@defer`
+has been used on a root mutation field:
+
+```raw graphql counter-example
+mutation {
+  ... @defer {
+    mutationField
+  }
+}
+```
+
+### Defer And Stream Directive Labels Are Unique
+
+** Formal Specification **
+
+- For every {directive} in a document.
+- Let {directiveName} be the name of {directive}.
+- If {directiveName} is "defer" or "stream":
+  - For every {argument} in {directive}:
+    - Let {argumentName} be the name of {argument}.
+    - Let {argumentValue} be the value passed to {argument}.
+    - If {argumentName} is "label":
+      - {argumentValue} must not be a variable.
+    - Let {labels} be all label values passed to defer or stream directive label
+      arguments.
+    - {labels} must be a set of one.
```

Comment:
that's not clear - set of one? shouldn't it be list with no repetitions?

Reply:
I think in other rules it is different situation. The phrase does not make sense here in this algorithm. What it says now is that there must be only ONE label in the request; while intent I think there might be several, but no repetitions

Reply:
I updated it to explicitly check for label values against a set of previously encountered values

Reply:
still does not make sense. It says 'only one label is allowed in the entire request'

Reply:
now it looks right